modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -421,66 +421,142 @@ public PathElement[] getCaretShape(int offset, boolean isLeading,
     }
 
     @Override
-    public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRunStart) {
+    public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRunStart, boolean forTextFlow) {
+        boolean leading = false;
+        boolean isMirrored = isMirrored(); // Node orientation is RTL
         int charIndex = -1;
         int insertionIndex = -1;
-        boolean leading = false;
         int relIndex = 0;
+        int ltrIndex = 0;
         int textWidthPrevLine = 0;
+        float xHitPos = x;
 
         ensureLayout();
         int lineIndex = getLineIndex(y, text, curRunStart);
         if (lineIndex >= getLineCount()) {
             charIndex = getCharCount();
             insertionIndex = charIndex + 1;
         } else {
-            if (isMirrored()) {
+            if (isMirrored && (forTextFlow || spans == null)) {
                 x = getMirroringWidth() - x;
             }
             TextLine line = lines[lineIndex];
             TextRun[] runs = line.getRuns();
             RectBounds bounds = line.getBounds();
             TextRun run = null;
-            x -= bounds.getMinX();
             //TODO binary search
             if (text == null || spans == null) {
```

Comment:
As I mentioned earlier, the condition is required to differentiate between Text, TexFlow and Text embedded in TextFlow. Hence not removing condition, made modification to use `forTextFlow` instead of `text == null`